apps/store/src/services/Tracking/Tracking.ts

```
@@ -0,0 +1,89 @@
+import { ProductOfferFragment } from '@/services/apollo/generated'
+import { AppTrackingContext, pushToGTMDataLayer, setGtmContext } from '@/services/gtm'
+import { isBrowser } from '@/utils/env'
+import { newSiteAbTest } from '../../newSiteAbTest'
+
+type TrackingContext = Record<string, unknown>
+
+export enum TrackingEvent {
+  ExperimentImpression = 'experiment_impression',
+  OfferCreated = 'offer_created',
+  PageView = 'virtual_page_view',
+}
+
+// Simple version with 2 destinations (GTM and Datadog) implemented inline
+export class Tracking {
+  constructor(public context: TrackingContext = {}) {}
+
+  public setContext = (key: string, value: unknown) => {
+    if (this.context[key] !== value) {
+      console.debug('tracking context', key, value)
+    }
+    this.context[key] = value
+  }
+
+  public setAppContext = (context: AppTrackingContext) => {
+    this.setContext('countryCode', context.countryCode)
+    setGtmContext(context)
+  }
+
+  public reportPageView(urlPath: string) {
+    this.ensureBrowserEnvironment()
```

Should we really throw an error here? Would be really bad UX to crash the app just because we setup analytics correctly, no?

---

Assuming we call analytics only from event handlers, this should never be called server-side.  If it does, it's programmer error and sooner we crash, sooner we detect and fix it